### PR TITLE
Cloudp 84351 add agent images to dockerfile generator

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

This PR moves the static agent Dockerfiles into templates for the dockerfile_generator

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [n/a] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: scripts/dev/dockerfile_generator.py [hunk 1]
```diff
@@ -9,6 +11,43 @@
 GOLANG_TAG = "1.14"
 
 
+def _shared_agent_params() -> DockerParameters:
+    with open("release.json", "r") as f:
+        release = json.loads(f.read())
+
+    return {
+        "template_path": "scripts/dev/templates/agent",
```
Comment (bznein) [inline scripts/dev/dockerfile_generator.py:19]: [q] why is this is shared if we always replace that?

**Review by bznein — APPROVED** [review]
LGTM only one q!

File: scripts/dev/dockerfile_generator.py (hunk 1 above)
Comment (chatton) [reply scripts/dev/dockerfile_generator.py:19]: both agents have the same template path, but currently all the other images are just in "scripts/dev/templates"

Comment (chatton) [reply scripts/dev/dockerfile_generator.py:19]: ah just seeing now you mean we replace it in both but with the same value - this is a mistake! Will correct it.

**Pull request merged**